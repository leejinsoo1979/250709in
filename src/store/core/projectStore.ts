import { create } from 'zustand';

// 기본 정보 타입
export interface BasicInfo {
  title: string;
  location: string;
}

// 프로젝트 상태 타입
interface ProjectState {
  // 상태
  basicInfo: BasicInfo;
  isDirty: boolean;  // 변경사항 있음을 표시
  
  // 기본 정보 액션
  setBasicInfo: (info: Partial<BasicInfo>) => void;
  resetBasicInfo: () => void;
  
  // 전체 상태 관리
  resetAll: () => void;
  markAsSaved: () => void;
}

// 초기 상태
const initialState: Omit<ProjectState, 'setBasicInfo' | 'resetBasicInfo' | 'resetAll' | 'markAsSaved'> = {
  basicInfo: {
    title: '',
    location: '',
  },
  isDirty: false,
};

export const useProjectStore = create<ProjectState>()((set) => ({
  ...initialState,
  
  // 기본 정보 설정
  setBasicInfo: (info) =>
    set((state) => ({
      basicInfo: { ...state.basicInfo, ...info },
      isDirty: true,
    })),
  
  // 기본 정보 초기화
  resetBasicInfo: () =>
    set({
      basicInfo: initialState.basicInfo,
      isDirty: true,
    }),
  
  // 전체 상태 초기화
  resetAll: () => set({ ...initialState, isDirty: false }),
  
  // 저장 상태로 마킹
  markAsSaved: () => set({ isDirty: false }),
})); 