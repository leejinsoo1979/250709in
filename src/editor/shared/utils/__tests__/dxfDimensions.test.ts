import { describe, it, expect } from 'vitest';
import { generateDXF } from '../dxfGenerator';
import { SpaceInfo } from '@/store/core/spaceConfigStore';

describe('[VALIDATOR] DXF Dimensions Format Test - WxHxD Pattern Verification', () => {
  // Test space configuration
  const testSpaceInfo: SpaceInfo = {
    width: 4000,
    height: 2400,
    depth: 600,
    baseConfig: {
      type: 'base_frame',
      height: 100,
    },
    backConfig: {
      type: 'back_panel',
      thickness: 18,
    },
    topConfig: {
      type: 'none',
    }
  };

  describe('Dimension Pattern Validation', () => {
    it('‚úÖ SINGLE furniture: Should generate DXF with correct WxHxD pattern', () => {
      // Single furniture sample (Ïò§ÌîàÎ∞ïÏä§)
      const singleModule = {
        id: 'single-1',
        moduleId: 'open-box',
        position: { x: 0, y: 0, z: 0 },
        moduleData: {
          name: 'Open Box',
          dimensions: {
            width: 800,
            height: 400,
            depth: 580
          }
        }
      };

      const dxfString = generateDXF({
        spaceInfo: testSpaceInfo,
        placedModules: [singleModule],
        drawingType: 'front'
      });

      // Verify WxHxD pattern (800W x 400H x 580D)
      const expectedPattern = '800W x 400H x 580D';
      const containsCorrectPattern = dxfString.includes(expectedPattern);
      
      // Check for incorrect HxD swap patterns
      const incorrectPattern1 = '800W x 580H x 400D'; // H and D swapped
      const incorrectPattern2 = '800 x 400 x 580'; // Missing W, H, D labels
      const hasIncorrectPattern = dxfString.includes(incorrectPattern1) || dxfString.includes(incorrectPattern2);

      expect(containsCorrectPattern).toBe(true);
      expect(hasIncorrectPattern).toBe(false);

      // Snapshot test for regression
      expect(dxfString).toMatchSnapshot('single-furniture-dxf');
      
      console.log('[VALIDATOR] ‚úÖ SINGLE furniture test passed - WxHxD pattern correct');
    });

    it('‚úÖ DUAL furniture: Should generate DXF with correct WxHxD pattern', () => {
      // Dual furniture sample (ÎìÄÏñº 2Îã®)
      const dualModule = {
        id: 'dual-1',
        moduleId: 'dual-2-shelf',
        position: { x: 0, y: 0, z: 0 },
        moduleData: {
          name: 'Dual 2-Shelf',
          dimensions: {
            width: 1600,
            height: 800,
            depth: 580
          }
        },
        isDualSlot: true
      };

      const dxfString = generateDXF({
        spaceInfo: testSpaceInfo,
        placedModules: [dualModule],
        drawingType: 'front'
      });

      // Verify WxHxD pattern (1600W x 800H x 580D)
      const expectedPattern = '1600W x 800H x 580D';
      const containsCorrectPattern = dxfString.includes(expectedPattern);
      
      // Check for incorrect HxD swap patterns
      const incorrectPattern1 = '1600W x 580H x 800D'; // H and D swapped
      const incorrectPattern2 = '1600 x 800 x 580'; // Missing W, H, D labels
      const hasIncorrectPattern = dxfString.includes(incorrectPattern1) || dxfString.includes(incorrectPattern2);

      expect(containsCorrectPattern).toBe(true);
      expect(hasIncorrectPattern).toBe(false);

      // Snapshot test for regression
      expect(dxfString).toMatchSnapshot('dual-furniture-dxf');
      
      console.log('[VALIDATOR] ‚úÖ DUAL furniture test passed - WxHxD pattern correct');
    });

    it('‚úÖ ASYMMETRIC furniture: Should generate DXF with correct WxHxD pattern', () => {
      // Asymmetric furniture sample (7Îã® - tall and narrow)
      const asymmetricModule = {
        id: 'asymmetric-1',
        moduleId: '7-shelf',
        position: { x: 0, y: 0, z: 0 },
        moduleData: {
          name: '7-Shelf',
          dimensions: {
            width: 800,
            height: 2100,  // Much taller than it is deep
            depth: 580
          }
        }
      };

      const dxfString = generateDXF({
        spaceInfo: testSpaceInfo,
        placedModules: [asymmetricModule],
        drawingType: 'front'
      });

      // Verify WxHxD pattern (800W x 2100H x 580D)
      const expectedPattern = '800W x 2100H x 580D';
      const containsCorrectPattern = dxfString.includes(expectedPattern);
      
      // Check for incorrect HxD swap patterns
      const incorrectPattern1 = '800W x 580H x 2100D'; // H and D swapped - CRITICAL ERROR
      const incorrectPattern2 = '800W x 2100D x 580H'; // D and H positions swapped
      const incorrectPattern3 = '800 x 2100 x 580'; // Missing W, H, D labels
      const hasIncorrectPattern = 
        dxfString.includes(incorrectPattern1) || 
        dxfString.includes(incorrectPattern2) || 
        dxfString.includes(incorrectPattern3);

      expect(containsCorrectPattern).toBe(true);
      expect(hasIncorrectPattern).toBe(false);

      // Additional validation: Height should be much larger than Depth
      const heightMatch = dxfString.match(/(\d+)H/);
      const depthMatch = dxfString.match(/(\d+)D/);
      
      if (heightMatch && depthMatch) {
        const height = parseInt(heightMatch[1]);
        const depth = parseInt(depthMatch[1]);
        expect(height).toBeGreaterThan(depth); // Height (2100) should be > Depth (580)
      }

      // Snapshot test for regression
      expect(dxfString).toMatchSnapshot('asymmetric-furniture-dxf');
      
      console.log('[VALIDATOR] ‚úÖ ASYMMETRIC furniture test passed - WxHxD pattern correct');
    });

    it('üîç VALIDATION SUMMARY: All dimension patterns should follow WxHxD format', () => {
      const modules = [
        {
          id: 'test-1',
          moduleId: 'test-module',
          position: { x: -800, y: 0, z: 0 },
          moduleData: {
            name: 'Test Module 1',
            dimensions: { width: 800, height: 1200, depth: 400 }
          }
        },
        {
          id: 'test-2',
          moduleId: 'test-module-2',
          position: { x: 800, y: 0, z: 0 },
          moduleData: {
            name: 'Test Module 2',
            dimensions: { width: 1600, height: 600, depth: 500 }
          }
        }
      ];

      const dxfString = generateDXF({
        spaceInfo: testSpaceInfo,
        placedModules: modules,
        drawingType: 'front'
      });

      // Extract all dimension patterns
      const dimensionPattern = /(\d+)W x (\d+)H x (\d+)D/g;
      const matches = [...dxfString.matchAll(dimensionPattern)];

      // Validate each match
      expect(matches.length).toBeGreaterThan(0);
      
      matches.forEach((match, index) => {
        const [full, width, height, depth] = match;
        const module = modules[index];
        
        if (module) {
          expect(parseInt(width)).toBe(module.moduleData.dimensions.width);
          expect(parseInt(height)).toBe(module.moduleData.dimensions.height);
          expect(parseInt(depth)).toBe(module.moduleData.dimensions.depth);
          
          console.log(`[VALIDATOR] ‚úÖ Module ${index + 1}: ${full} - Pattern verified`);
        }
      });

      // Check for any incorrect patterns
      const incorrectPatterns = [
        /(\d+)W x (\d+)D x (\d+)H/g,  // D and H swapped
        /(\d+) x (\d+) x (\d+)/g,      // Missing labels
        /(\d+)H x (\d+)W x (\d+)D/g,  // Wrong order
      ];

      incorrectPatterns.forEach((pattern, index) => {
        const incorrectMatches = [...dxfString.matchAll(pattern)];
        // Allow matches only if they're part of the correct pattern
        incorrectMatches.forEach(match => {
          const matchStr = match[0];
          // Check if this is actually part of a correct WxHxD pattern
          if (!matchStr.includes('W x') || !matchStr.includes('H x') || !matchStr.includes('D')) {
            expect(incorrectMatches.length).toBe(0);
          }
        });
      });

      console.log('[VALIDATOR] ‚úÖ All dimension patterns validated successfully');
    });
  });

  describe('Pattern Failure Detection', () => {
    it('‚ùå Should FAIL if HxD values are swapped', () => {
      // This test validates that we can detect when H and D are incorrectly swapped
      const testModule = {
        id: 'fail-test',
        moduleId: 'test-module',
        position: { x: 0, y: 0, z: 0 },
        moduleData: {
          name: 'Test Module',
          dimensions: {
            width: 1000,
            height: 1500, // Height is 1500
            depth: 400    // Depth is 400
          }
        }
      };

      const dxfString = generateDXF({
        spaceInfo: testSpaceInfo,
        placedModules: [testModule],
        drawingType: 'front'
      });

      // The correct pattern should be 1000W x 1500H x 400D
      const correctPattern = '1000W x 1500H x 400D';
      const swappedPattern = '1000W x 400H x 1500D'; // WRONG - H and D values swapped

      // Verify correct pattern exists
      expect(dxfString.includes(correctPattern)).toBe(true);
      // Verify swapped pattern does NOT exist
      expect(dxfString.includes(swappedPattern)).toBe(false);

      if (dxfString.includes(swappedPattern)) {
        console.error('[VALIDATOR] ‚ùå CRITICAL ERROR: H and D values are swapped!');
        throw new Error('Dimension format validation failed: H and D values are swapped');
      } else {
        console.log('[VALIDATOR] ‚úÖ H and D swap detection test passed');
      }
    });
  });
});