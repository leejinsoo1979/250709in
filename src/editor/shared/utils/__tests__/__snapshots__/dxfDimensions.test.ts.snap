// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[VALIDATOR] DXF Dimensions Format Test - WxHxD Pattern Verification > Dimension Pattern Validation > ✅ ASYMMETRIC furniture: Should generate DXF with correct WxHxD pattern > asymmetric-furniture-dxf 1`] = `
"0
SECTION
2
HEADER
9
$ACADVER
1
AC1021
9
$LASTSAVEDBY
1
@tarikjabiri/dxf
9
$HANDSEED
5
28
9
$INSUNITS
70
0
9
$CLAYER
8
TEXT
9
$VIEWCTR
10
1240.25
20
950
0
ENDSEC
0
SECTION
2
CLASSES
0
ENDSEC
0
SECTION
2
TABLES
0
TABLE
2
VPORT
5
1
330
0
100
AcDbSymbolTable
70
1
0
VPORT
5
19
330
1
100
AcDbSymbolTableRecord
100
AcDbViewportTableRecord
2
*Active
70
0
10
0
20
0
11
1
21
1
12
1240.25
22
950
13
0
23
0
14
10
24
10
15
10
25
10
16
0
26
0
36
1
17
0
27
0
37
0
40
2500
41
2
42
50
43
0
44
0
50
0
51
0
71
0
72
100
73
1
74
3
75
0
76
1
77
0
78
0
281
0
65
1
110
0
120
0
130
0
111
1
121
0
131
0
112
0
122
1
132
0
79
0
146
0
348
10020
60
7
61
5
292
1
282
1
141
0
142
0
63
250
421
3358443
0
ENDTAB
0
TABLE
2
LTYPE
5
2
330
0
100
AcDbSymbolTable
70
3
0
LTYPE
5
12
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByBlock
70
0
3

72
65
73
0
40
0
0
LTYPE
5
13
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByLayer
70
0
3

72
65
73
0
40
0
0
LTYPE
5
14
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
Continuous
70
0
3
Solid line
72
65
73
0
40
0
0
ENDTAB
0
TABLE
2
LAYER
5
3
330
0
100
AcDbSymbolTable
70
4
0
LAYER
5
15
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
0
70
0
62
7
6
Continuous
370
0
390
0
0
LAYER
5
1A
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
FURNITURE
70
0
62
3
6
Continuous
370
0
390
0
0
LAYER
5
1B
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
DIMENSIONS
70
0
62
1
6
Continuous
370
0
390
0
0
LAYER
5
1C
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
TEXT
70
0
62
5
6
Continuous
370
0
390
0
0
ENDTAB
0
TABLE
2
STYLE
5
4
330
0
100
AcDbSymbolTable
70
1
0
STYLE
5
16
330
4
100
AcDbSymbolTableRecord
100
AcDbTextStyleTableRecord
2
Standard
70
0
40
0
41
1
50
0
71
0
42
1
3
txt
4

0
ENDTAB
0
TABLE
2
VIEW
5
5
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
UCS
5
6
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
APPID
5
7
330
0
100
AcDbSymbolTable
70
1
0
APPID
5
17
330
7
100
AcDbSymbolTableRecord
100
AcDbRegAppTableRecord
2
ACAD
70
0
0
ENDTAB
0
TABLE
2
DIMSTYLE
5
8
330
0
100
AcDbSymbolTable
70
1
100
AcDbDimStyleTable
0
DIMSTYLE
105
18
330
8
100
AcDbSymbolTableRecord
100
AcDbDimStyleTableRecord
2
Standard
70
0
340
16
0
ENDTAB
0
TABLE
2
BLOCK_RECORD
5
9
330
0
100
AcDbSymbolTable
70
2
0
BLOCK_RECORD
5
C
330
9
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Model_Space
70
0
280
1
281
0
0
BLOCK_RECORD
5
F
330
9
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Paper_Space
70
0
280
1
281
0
0
ENDTAB
0
ENDSEC
0
SECTION
2
BLOCKS
0
BLOCK
5
D
330
C
100
AcDbEntity
8
TEXT
100
AcDbBlockBegin
2
*Model_Space
70
0
10
0
20
0
30
0
3
*Model_Space
1

0
ENDBLK
5
E
100
AcDbEntity
8
0
100
AcDbBlockEnd
0
BLOCK
5
10
330
F
100
AcDbEntity
8
0
100
AcDbBlockBegin
2
*Paper_Space
70
0
10
0
20
0
30
0
3
*Paper_Space
1

0
ENDBLK
5
11
100
AcDbEntity
8
0
100
AcDbBlockEnd
0
ENDSEC
0
SECTION
2
ENTITIES
0
LINE
5
1D
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
1600.5
20
100
30
0
11
2400.5
21
100
31
0
0
LINE
5
1E
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
2400.5
20
100
30
0
11
2400.5
21
2200
31
0
0
LINE
5
1F
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
2400.5
20
2200
30
0
11
1600.5
21
2200
31
0
0
LINE
5
20
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
1600.5
20
2200
30
0
11
1600.5
21
100
31
0
0
TEXT
5
21
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2000.5
20
1150
30
0
40
40
1
7-Shelf
7
STANDARD
100
AcDbText
0
TEXT
5
22
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2000.5
20
40
30
0
40
20
1
800W x 2100H x 580D
7
STANDARD
100
AcDbText
0
LINE
5
23
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2450.5
20
100
30
0
11
2450.5
21
2200
31
0
0
LINE
5
24
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2430.5
20
100
30
0
11
2470.5
21
100
31
0
0
LINE
5
25
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2430.5
20
2200
30
0
11
2470.5
21
2200
31
0
0
TEXT
5
26
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2480.5
20
1150
30
0
40
20
1
2100mm
7
STANDARD
100
AcDbText
0
TEXT
5
27
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
0
20
-200
30
0
40
60
1
Front Elevation - Furniture Layout
7
STANDARD
100
AcDbText
0
ENDSEC
0
SECTION
2
OBJECTS
0
DICTIONARY
5
A
330
0
100
AcDbDictionary
281
1
3
ACAD_GROUP
350
B
0
DICTIONARY
5
B
330
A
100
AcDbDictionary
281
0
0
ENDSEC
0
EOF"
`;

exports[`[VALIDATOR] DXF Dimensions Format Test - WxHxD Pattern Verification > Dimension Pattern Validation > ✅ DUAL furniture: Should generate DXF with correct WxHxD pattern > dual-furniture-dxf 1`] = `
"0
SECTION
2
HEADER
9
$ACADVER
1
AC1021
9
$LASTSAVEDBY
1
@tarikjabiri/dxf
9
$HANDSEED
5
28
9
$INSUNITS
70
0
9
$CLAYER
8
TEXT
9
$VIEWCTR
10
1440.25
20
300
0
ENDSEC
0
SECTION
2
CLASSES
0
ENDSEC
0
SECTION
2
TABLES
0
TABLE
2
VPORT
5
1
330
0
100
AcDbSymbolTable
70
1
0
VPORT
5
19
330
1
100
AcDbSymbolTableRecord
100
AcDbViewportTableRecord
2
*Active
70
0
10
0
20
0
11
1
21
1
12
1440.25
22
300
13
0
23
0
14
10
24
10
15
10
25
10
16
0
26
0
36
1
17
0
27
0
37
0
40
1200
41
2
42
50
43
0
44
0
50
0
51
0
71
0
72
100
73
1
74
3
75
0
76
1
77
0
78
0
281
0
65
1
110
0
120
0
130
0
111
1
121
0
131
0
112
0
122
1
132
0
79
0
146
0
348
10020
60
7
61
5
292
1
282
1
141
0
142
0
63
250
421
3358443
0
ENDTAB
0
TABLE
2
LTYPE
5
2
330
0
100
AcDbSymbolTable
70
3
0
LTYPE
5
12
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByBlock
70
0
3

72
65
73
0
40
0
0
LTYPE
5
13
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByLayer
70
0
3

72
65
73
0
40
0
0
LTYPE
5
14
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
Continuous
70
0
3
Solid line
72
65
73
0
40
0
0
ENDTAB
0
TABLE
2
LAYER
5
3
330
0
100
AcDbSymbolTable
70
4
0
LAYER
5
15
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
0
70
0
62
7
6
Continuous
370
0
390
0
0
LAYER
5
1A
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
FURNITURE
70
0
62
3
6
Continuous
370
0
390
0
0
LAYER
5
1B
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
DIMENSIONS
70
0
62
1
6
Continuous
370
0
390
0
0
LAYER
5
1C
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
TEXT
70
0
62
5
6
Continuous
370
0
390
0
0
ENDTAB
0
TABLE
2
STYLE
5
4
330
0
100
AcDbSymbolTable
70
1
0
STYLE
5
16
330
4
100
AcDbSymbolTableRecord
100
AcDbTextStyleTableRecord
2
Standard
70
0
40
0
41
1
50
0
71
0
42
1
3
txt
4

0
ENDTAB
0
TABLE
2
VIEW
5
5
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
UCS
5
6
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
APPID
5
7
330
0
100
AcDbSymbolTable
70
1
0
APPID
5
17
330
7
100
AcDbSymbolTableRecord
100
AcDbRegAppTableRecord
2
ACAD
70
0
0
ENDTAB
0
TABLE
2
DIMSTYLE
5
8
330
0
100
AcDbSymbolTable
70
1
100
AcDbDimStyleTable
0
DIMSTYLE
105
18
330
8
100
AcDbSymbolTableRecord
100
AcDbDimStyleTableRecord
2
Standard
70
0
340
16
0
ENDTAB
0
TABLE
2
BLOCK_RECORD
5
9
330
0
100
AcDbSymbolTable
70
2
0
BLOCK_RECORD
5
C
330
9
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Model_Space
70
0
280
1
281
0
0
BLOCK_RECORD
5
F
330
9
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Paper_Space
70
0
280
1
281
0
0
ENDTAB
0
ENDSEC
0
SECTION
2
BLOCKS
0
BLOCK
5
D
330
C
100
AcDbEntity
8
TEXT
100
AcDbBlockBegin
2
*Model_Space
70
0
10
0
20
0
30
0
3
*Model_Space
1

0
ENDBLK
5
E
100
AcDbEntity
8
0
100
AcDbBlockEnd
0
BLOCK
5
10
330
F
100
AcDbEntity
8
0
100
AcDbBlockBegin
2
*Paper_Space
70
0
10
0
20
0
30
0
3
*Paper_Space
1

0
ENDBLK
5
11
100
AcDbEntity
8
0
100
AcDbBlockEnd
0
ENDSEC
0
SECTION
2
ENTITIES
0
LINE
5
1D
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
1200.5
20
100
30
0
11
2800.5
21
100
31
0
0
LINE
5
1E
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
2800.5
20
100
30
0
11
2800.5
21
900
31
0
0
LINE
5
1F
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
2800.5
20
900
30
0
11
1200.5
21
900
31
0
0
LINE
5
20
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
1200.5
20
900
30
0
11
1200.5
21
100
31
0
0
TEXT
5
21
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2000.5
20
500
30
0
40
40
1
Dual 2-Shelf
7
STANDARD
100
AcDbText
0
TEXT
5
22
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2000.5
20
40
30
0
40
20
1
1600W x 800H x 580D
7
STANDARD
100
AcDbText
0
LINE
5
23
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2850.5
20
100
30
0
11
2850.5
21
900
31
0
0
LINE
5
24
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2830.5
20
100
30
0
11
2870.5
21
100
31
0
0
LINE
5
25
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2830.5
20
900
30
0
11
2870.5
21
900
31
0
0
TEXT
5
26
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2880.5
20
500
30
0
40
20
1
800mm
7
STANDARD
100
AcDbText
0
TEXT
5
27
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
0
20
-200
30
0
40
60
1
Front Elevation - Furniture Layout
7
STANDARD
100
AcDbText
0
ENDSEC
0
SECTION
2
OBJECTS
0
DICTIONARY
5
A
330
0
100
AcDbDictionary
281
1
3
ACAD_GROUP
350
B
0
DICTIONARY
5
B
330
A
100
AcDbDictionary
281
0
0
ENDSEC
0
EOF"
`;

exports[`[VALIDATOR] DXF Dimensions Format Test - WxHxD Pattern Verification > Dimension Pattern Validation > ✅ SINGLE furniture: Should generate DXF with correct WxHxD pattern > single-furniture-dxf 1`] = `
"0
SECTION
2
HEADER
9
$ACADVER
1
AC1021
9
$LASTSAVEDBY
1
@tarikjabiri/dxf
9
$HANDSEED
5
28
9
$INSUNITS
70
0
9
$CLAYER
8
TEXT
9
$VIEWCTR
10
1240.25
20
100
0
ENDSEC
0
SECTION
2
CLASSES
0
ENDSEC
0
SECTION
2
TABLES
0
TABLE
2
VPORT
5
1
330
0
100
AcDbSymbolTable
70
1
0
VPORT
5
19
330
1
100
AcDbSymbolTableRecord
100
AcDbViewportTableRecord
2
*Active
70
0
10
0
20
0
11
1
21
1
12
1240.25
22
100
13
0
23
0
14
10
24
10
15
10
25
10
16
0
26
0
36
1
17
0
27
0
37
0
40
800
41
2
42
50
43
0
44
0
50
0
51
0
71
0
72
100
73
1
74
3
75
0
76
1
77
0
78
0
281
0
65
1
110
0
120
0
130
0
111
1
121
0
131
0
112
0
122
1
132
0
79
0
146
0
348
10020
60
7
61
5
292
1
282
1
141
0
142
0
63
250
421
3358443
0
ENDTAB
0
TABLE
2
LTYPE
5
2
330
0
100
AcDbSymbolTable
70
3
0
LTYPE
5
12
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByBlock
70
0
3

72
65
73
0
40
0
0
LTYPE
5
13
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByLayer
70
0
3

72
65
73
0
40
0
0
LTYPE
5
14
330
2
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
Continuous
70
0
3
Solid line
72
65
73
0
40
0
0
ENDTAB
0
TABLE
2
LAYER
5
3
330
0
100
AcDbSymbolTable
70
4
0
LAYER
5
15
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
0
70
0
62
7
6
Continuous
370
0
390
0
0
LAYER
5
1A
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
FURNITURE
70
0
62
3
6
Continuous
370
0
390
0
0
LAYER
5
1B
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
DIMENSIONS
70
0
62
1
6
Continuous
370
0
390
0
0
LAYER
5
1C
330
3
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
TEXT
70
0
62
5
6
Continuous
370
0
390
0
0
ENDTAB
0
TABLE
2
STYLE
5
4
330
0
100
AcDbSymbolTable
70
1
0
STYLE
5
16
330
4
100
AcDbSymbolTableRecord
100
AcDbTextStyleTableRecord
2
Standard
70
0
40
0
41
1
50
0
71
0
42
1
3
txt
4

0
ENDTAB
0
TABLE
2
VIEW
5
5
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
UCS
5
6
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
APPID
5
7
330
0
100
AcDbSymbolTable
70
1
0
APPID
5
17
330
7
100
AcDbSymbolTableRecord
100
AcDbRegAppTableRecord
2
ACAD
70
0
0
ENDTAB
0
TABLE
2
DIMSTYLE
5
8
330
0
100
AcDbSymbolTable
70
1
100
AcDbDimStyleTable
0
DIMSTYLE
105
18
330
8
100
AcDbSymbolTableRecord
100
AcDbDimStyleTableRecord
2
Standard
70
0
340
16
0
ENDTAB
0
TABLE
2
BLOCK_RECORD
5
9
330
0
100
AcDbSymbolTable
70
2
0
BLOCK_RECORD
5
C
330
9
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Model_Space
70
0
280
1
281
0
0
BLOCK_RECORD
5
F
330
9
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Paper_Space
70
0
280
1
281
0
0
ENDTAB
0
ENDSEC
0
SECTION
2
BLOCKS
0
BLOCK
5
D
330
C
100
AcDbEntity
8
TEXT
100
AcDbBlockBegin
2
*Model_Space
70
0
10
0
20
0
30
0
3
*Model_Space
1

0
ENDBLK
5
E
100
AcDbEntity
8
0
100
AcDbBlockEnd
0
BLOCK
5
10
330
F
100
AcDbEntity
8
0
100
AcDbBlockBegin
2
*Paper_Space
70
0
10
0
20
0
30
0
3
*Paper_Space
1

0
ENDBLK
5
11
100
AcDbEntity
8
0
100
AcDbBlockEnd
0
ENDSEC
0
SECTION
2
ENTITIES
0
LINE
5
1D
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
1600.5
20
100
30
0
11
2400.5
21
100
31
0
0
LINE
5
1E
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
2400.5
20
100
30
0
11
2400.5
21
500
31
0
0
LINE
5
1F
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
2400.5
20
500
30
0
11
1600.5
21
500
31
0
0
LINE
5
20
330
C
100
AcDbEntity
8
FURNITURE
100
AcDbLine
10
1600.5
20
500
30
0
11
1600.5
21
100
31
0
0
TEXT
5
21
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2000.5
20
300
30
0
40
40
1
Open Box
7
STANDARD
100
AcDbText
0
TEXT
5
22
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2000.5
20
40
30
0
40
20
1
800W x 400H x 580D
7
STANDARD
100
AcDbText
0
LINE
5
23
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2450.5
20
100
30
0
11
2450.5
21
500
31
0
0
LINE
5
24
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2430.5
20
100
30
0
11
2470.5
21
100
31
0
0
LINE
5
25
330
C
100
AcDbEntity
8
DIMENSIONS
100
AcDbLine
10
2430.5
20
500
30
0
11
2470.5
21
500
31
0
0
TEXT
5
26
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
2480.5
20
300
30
0
40
20
1
400mm
7
STANDARD
100
AcDbText
0
TEXT
5
27
330
C
100
AcDbEntity
8
TEXT
100
AcDbText
10
0
20
-200
30
0
40
60
1
Front Elevation - Furniture Layout
7
STANDARD
100
AcDbText
0
ENDSEC
0
SECTION
2
OBJECTS
0
DICTIONARY
5
A
330
0
100
AcDbDictionary
281
1
3
ACAD_GROUP
350
B
0
DICTIONARY
5
B
330
A
100
AcDbDictionary
281
0
0
ENDSEC
0
EOF"
`;
